python/foglamp/common/configuration_manager.py

```diff
@@ -745,6 +744,72 @@ def get_entry_val(k):
                 'Unable to run callbacks for category_name %s', category_name)
             raise
 
+    async def set_optional_value_entry(self, category_name, item_name, optional_entry_name, new_value_entry):
+        try:
+            storage_value_entry = None
+            if category_name in self._cacheManager:
+                if item_name not in self._cacheManager.cache[category_name]['value']:
+                    raise ValueError("No detail found for the category_name: {} and item_name: {}"
+                                     .format(category_name, item_name))
+                storage_value_entry = self._cacheManager.cache[category_name]['value'][item_name]
+                if optional_entry_name not in storage_value_entry:
+                    raise KeyError("{} does not exist".format(optional_entry_name))
+                if storage_value_entry[optional_entry_name] == new_value_entry:
+                    return
+            else:
+                # get storage_value_entry and compare against new_value_value with its type, update if different
+                storage_value_entry = await self._read_item_val(category_name, item_name)
+                # check for category_name and item_name combination existence in storage
+                if storage_value_entry is None:
+                    raise ValueError("No detail found for the category_name: {} and item_name: {}"
+                                     .format(category_name, item_name))
+                if storage_value_entry[optional_entry_name] == new_value_entry:
+                    return
+            # Validate optional types
```

Comment:
Since "" is also a valid value, to unset an optional item, we need to put line#769-794 under "if new_value_entry != "":".

Reply:
Also, at line#711, the error message should be changed to:
`raise TypeError('Type `mismatch/Unrecognized` value name for item_name {}'.format(item_name))`